crates/accelerate/src/circuit_library/pauli_feature_map.rs

```diff
@@ -0,0 +1,317 @@
+        .chars()
+        .rev() // reverse due to Qiskit's bit ordering convention
+        .zip(qubits)
+        .filter(|(p, _)| *p != 'i')
+        .collect_vec();
+
+    // get the basis change: x -> HGate, y -> RXGate(pi/2), z -> nothing
+    let basis_change = active_paulis
+        .clone()
+        .into_iter()
+        .filter(|(p, _)| *p != 'z')
+        .map(|(p, q)| match p {
+            'x' => (StandardGate::HGate, smallvec![], smallvec![q]),
+            'y' => (
+                StandardGate::RXGate,
+                smallvec![Param::Float(PI2)],
+                smallvec![q],
+            ),
+            _ => unreachable!("Invalid Pauli string."), // "z" and "i" have been filtered out
+        });
+
+    // get the inverse basis change
+    let inverse_basis_change = basis_change.clone().map(|(gate, _, qubit)| match gate {
+        StandardGate::HGate => (gate, smallvec![], qubit),
+        StandardGate::RXGate => (gate, smallvec![Param::Float(-PI2)], qubit),
+        _ => unreachable!(),
+    });
+
+    // get the CX chain down to the target rotation qubit
+    let chain_down = active_paulis
+        .clone()
+        .into_iter()
+        .map(|(_, q)| q)
+        .tuple_windows() // iterates over (q[i], q[i+1]) windows
+        .map(|(ctrl, target)| (StandardGate::CXGate, smallvec![], smallvec![ctrl, target]));
+
+    // get the CX chain up (cannot use chain_down.rev since tuple_windows is not double ended)
+    let chain_up = active_paulis
+        .clone()
+        .into_iter()
+        .rev()
+        .map(|(_, q)| q)
+        .tuple_windows()
+        .map(|(target, ctrl)| (StandardGate::CXGate, smallvec![], smallvec![ctrl, target]));
+
+    // get the RZ gate on the last qubit
+    let last_qubit = active_paulis.last().unwrap().1;
+    let z_rotation = std::iter::once((
+        StandardGate::PhaseGate,
+        smallvec![time],
+        smallvec![last_qubit],
+    ));
+
+    // and finally chain everything together
+    basis_change
+        .chain(chain_down)
+        .chain(z_rotation)
+        .chain(chain_up)
+        .chain(inverse_basis_change)
+}
+
+/// Build a Pauli feature map circuit.
+///
+/// Args:
+///     feature_dimension: The feature dimension (i.e. the number of qubits).
+///     parameters: A parameter vector with ``feature_dimension`` elements. Taken as input
+///         here to avoid a call to Python constructing the vector.
+///     reps: The number of repetitions of Hadamard + evolution layers.
+///     entanglement: The entanglement, given as Python string or None (defaults to "full").
+///     paulis: The Pauli strings as list of strings or None (default to ["z", "zz"]).
+///     alpha: A float multiplier for rotation angles.
+///     insert_barriers: Whether to insert barriers in between the Hadamard and evolution layers.
+///     data_map_func: An accumulation function that takes as input a vector of parameters the
+///         current gate acts on and returns a scalar.
+///     
```

Comment:
`parameter_prefix` appears in the python code but not here. `parameters` appear here and not in the python code.

Reply:
it's a bit confusing that both the python and rust functions have the same name

Reply:
Yeah, the parameters are constructed Python-side because (1) we avoid calling Python once we're in Rust space and (2) I'm not even sure we can create `ParameterVector`s Rust side 🤔 

And I gave them the same name because they do the same thing 🙂 I know we often have `function_inner` on the Rust side but IMO that makes it look like it's some internal-only function -- even though they are perfectly fine to be called from the Rust interface 😄 but we can discuss this, if you want to change it 👍🏻